src/7_has_deep_checker.js

```diff
@@ -1,3 +1,45 @@
-module.exports = function hasValueDepth() {
-  // TODO Implement me.
+module.exports = hasValueDepth;
+
+function hasValueDepth(objct, val) {
+  // funkcja toTab wchodzi w zagniedżdżone obiekty i ich wartości przenosi do tabeli
+  function toTab(obj, tab) {
+    for (let key in obj) {
+      //sprawdzam czy wartość obiektu jest typu 'obiekt'
+      if (typeof(obj[key]) === 'object') {
+        toTab(obj[key], tab); // jeśli tak to uruchamiam funkcję toTab dla tego obiektu
+      } else {tab.push(obj[key]);} //jeśli nie to wartość jest zapisywana do tabeli
```

Comment:
popraw formatowanie 🌵 
```suggestion
      } else {
          tab.push(obj[key]);
      } //jeśli nie to wartość jest zapisywana do tabeli
```